src/main/kotlin/ru/buhinder/alcopartyservice/service/EventService.kt

```diff
@@ -77,21 +82,27 @@ class EventService(
             .flatMap { eventAlcoholicDaoFacade.delete(it) }
     }
 
-    fun getAllEvents(alcoholicId: UUID): Flux<FullEventResponse> {
-        return eventDaoFacade.findAllAndAlcoholicIsNotBanned(alcoholicId)
-            .map { conversionService.convert(it, EventResponse::class.java)!! }
-            .flatMap { res ->
-                val eventId = res.id
-                eventPhotoDaoFacade.findAllByEventId(eventId)
-                    .map { it.photoId }
-                    .collectList()
-                    .flatMap { photos ->
-                        eventAlcoholicDaoFacade.findAllByEventId(eventId)
-                            .map { it.alcoholicId }
-                            .collectList()
-                            .map { FullEventResponse(res, photos, it.toList()) }
-                    }
-            }
+    fun getAllEvents(alcoholicId: UUID, page: Int, pageSize: Int): Mono<AllEventsPageableResponse> {
+        return Mono.zip(
```

Comment:
All these nested calls look much too complicated

>return eventDaoFacade.findAllAndAlcoholicIsNotBanned(alcoholicId, page, pageSize)
.map { conversionService.convert(it, EventResponse::class.java)!! }
.flatMap { res ->
    val (eventId) = res
    eventAlcoholicDaoFacade.findAllByEventId(eventId).any { it.alcoholicId == alcoholicId }
        .flatMap { isParticipant ->
            eventPhotoDaoFacade.findFirstByEventId(alcoholicId).map { it.photoId }
                .map { SearchEventResponse(res, it, isParticipant) }
                .switchIfEmpty { SearchEventResponse(res, null, isParticipant).toMono() }
        }
}
.collectList()
.switchIfEmpty { emptyList<SearchEventResponse>().toMono() }
.zipWith(eventDaoFacade.countAllAndAlcoholicIsNotBanned(alcoholicId))
.map { AllEventsPageableResponse(it.t1, util.createPagination(it.t2, page, pageSize)) }